### ISSUE TITLE
OpenTracing Scope.span() has been removed.

### ISSUE THREAD

**Issue by ddeaguiar** [pedestal/pedestal#693]
Version [0.31](https://github.com/opentracing/opentracing-java#deprecated-members-since-031) of the opentracing api deprecated `Scope.span()` as well as other methods. Version [0.33](https://github.com/opentracing/opentracing-java/blob/master/CHANGELOG.md#v0330-2019-05-06) removed these deprecated methods. A number of the tracing protocol extensions to Scope in `io.pedestal.log` are impacted by this change and require analysis and resolution.

The reflection warnings that arise from requiring `io.pedestal.log` capture the affected implementations.

```
Reflection warning, io/pedestal/log.clj:687:26 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:689:16 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:693:20 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:696:20 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:736:17 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:738:17 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:741:19 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:743:19 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:773:21 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:775:21 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:799:19 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:801:26 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:802:36 - reference to field span on io.opentracing.Scope can't be resolved.
Reflection warning, io/pedestal/log.clj:803:43 - reference to field span on io.opentracing.Scope can't be resolved.
```

**Comment by ohpauleez** [comment]
fwiw, this is fixed in Pedestal2.

**Comment by gnarroway** [comment]
@ohpauleez are you referring to a major version bump of pedestal? If so is there anything written up about it? Very curious.

**Comment by ohpauleez** [comment]
Correct @gnarroway.  Current Pedestal should be considered Pedestal 1.0 - strong, stable, battle-tested.
Pedestal2 has been in active development since 2019 and I've been seeing how it works on production systems before opening it up to the wider community for participation.  There's nothing publicly written about it but that will hopefully change soon.
It's very similar in spirit, some namespaces have been reorganized (eg: all the servlet stuff is isolated to a single namespace, telemetry and logging are split, etc.).  There are some significant changes though -- Engines are a first-class thing, stored within the context (eg: If you don't like how async interceptors resume, you can define your own interceptor engine), the context can be any data type you want (you can use mutable Java HashMaps if you want), common cases were optimized (we've seen a Pedestal2 interceptor chain clear 1M reqs/s), Pedestal2 apps can be written in Java, and more control has been placed in the hands of users.

**Comment by christianromney** [comment]
I think we should start by looking at some analysis and design docs that clarify the problems Pedestal 2 aims to solve and how it addresses them. I remember @ohpauleez telling us about some exciting work he was doing but I don't know how many of us have visibility into those efforts since it's not been done on a branch in this repo which makes it hard to evaluate. To be completely honest, I'd be a little uncomfortable with a large push without any context.

**Comment by ohpauleez** [comment]
It's my plan to layout a document that looks at the problems within Pedestal1, the possible ways to address those problems along with their trade-offs, and supply the evaluation sandbox (the existing Pedestal2 code, very much as prototype).

**Comment by christianromney** [comment]
That sounds really great @ohpauleez. Sorry if I came across as a giant wet blanket, my intent is not to diminish your enthusiasm nor to be a roadblock. I just want to make sure the rest of the maintainers have the opportunity to collaborate with you meaningfully. Much love, brother.

**Comment by ddeaguiar** [comment]
> It's my plan to layout a document that looks at the problems within Pedestal1, the possible ways to address those problems along with their trade-offs, and supply the evaluation sandbox (the existing Pedestal2 code, very much as prototype).

Thanks @ohpauleez, looking forward to reading your take on this!

**Comment by ddeaguiar** [comment]
> fwiw, this is fixed in Pedestal2.

@ohpauleez can you expand on what you mean by "fixed" because it seems the the only path forward is to propagate the breaking change by removing `Scope` from the `TraceSpan`, `TraceSpanLog`, `TraceSpanLogMap` and `TraceSpanBaggage` protocols.

**Comment by ohpauleez** [comment]
Correct - Scope was removed from the protocols, the dependency was updated, following their notes on why the changes to Scope were made.

**Comment by jvia** [comment]
For now, should projects pin open tracing dependencies to `0.31`?

**Comment by ddeaguiar** [comment]
@jvia you could pin the open tracing dependencies to `0.32` if you desire but I think that would only be necessary if you use `Scope` in the way that has been deprecated and removed. If your tracing usage follows that of the Pedestal tracing sample where you create spans and log them directly, you should be ok.

**Comment by ddeaguiar** [comment]
Released in 0.5.10
